### ISSUE TITLE
Anybody port this to TensorFlow or Pytorch?

### ISSUE THREAD

**Issue by austinmw** [ChihebTrabelsi/deep_complex_networks#27]
Theano and Kerosene are no longer maintained which I think is preventing this project from widespread adoption. I Actually can't get Kerosene installed at all. Has anybody ported this to a maintained framework?

**Comment by harritaylor** [comment]
Hi, has there been any updates on this? I'd be happy to contribute if help is needed.

**Comment by chris1992212** [comment]
Has there been any updates now? I also want to transform it to Tensorflow, but I'm not sure whether I've done it right.

**Comment by planewave** [comment]
Same here, waiting for TensorFlow-Keras or PyTorch port

**Comment by JesperDramsch** [comment]
https://github.com/zsh965866221/Complex-Valued_Networks

Here's a keras implementation.

**Comment by lintongtong123** [comment]
mark, waiting for TensorFlow-Keras or PyTorch port

**Comment by veritas9872** [comment]
@lintongtong123 @planewave @harritaylor @obilaniu There is a very (extremely) basic implementation in Pytorch [here](https://github.com/litcoderr/ComplexCNN).
@chris1992212 Also, a slightly less basic implementation in Tensorflow was made [here](https://github.com/CedricChing/DeepMRI/blob/master/model.py).
I recall that there was another implementation in Keras using Tensorflow as its backend but I could not find the repo.
However, I think that it would be better to start from scratch and just copy some code from those two repositories as they are so basic. Only 2D convolutions are implemented and the Pytorch implementation does not even have proper weight initialization.

**Comment by JiachuanDENG** [comment]
conv2d part is trivial, Batch normalization should be the most difficult part to implement

**Comment by JesperDramsch** [comment]
I have a keras with Tensorflow fork of this. But it's mostly untested. I'd be happy to have contributions to it. https://github.com/JesperDramsch/keras-complex

It implements Conv1D to Con3D, Transpose, BN, Dense layers, Spectral Pooling. But it's very much research code from a non-CS major. Happy about help to get the standard up to par.
It's also TF1.x not TF2, so there's that.
